standard_revisions/match_variables_from_two_versions_of_a_dataset.py

```diff
@@ -0,0 +1,313 @@
+        print("\nAll variables in the new dataset have been matched.")
+
+
+def save_variable_replacements_file(mapping, output_file=OUTPUT_FILE):
+    """Save a json file with the mapping from old to new variable ids.
+
+    Parameters
+    ----------
+    mapping : pd.DataFrame
+        Mapping table from old variable name and id to new variable name and id.
+    output_file : str
+        Path to output file.
+
+    """
+    # Create a dictionary mapping from old variable id to new variable id.
+    mapping_indexes = mapping[['id_old', 'id_new']].set_index('id_old').to_dict()['id_new']
+    mapping_indexes = {str(key): str(mapping_indexes[key]) for key in mapping_indexes}
+
+    print(f"Saving index mapping to json file: {output_file}")
+    save_data_to_json_file(data=mapping_indexes, json_file=output_file, **{'indent': 4, 'sort_keys': True})
+
+
+def main(old_dataset_name, new_dataset_name, omit_identical=OMIT_IDENTICAL, matching_function=MATCHING_FUNCTION,
+         output_file=OUTPUT_FILE):
+    with get_connection() as db_conn:
+        # Get old and new dataset ids.
+        old_dataset_id = get_dataset_id(db_conn=db_conn, dataset_name=old_dataset_name)
+        new_dataset_id = get_dataset_id(db_conn=db_conn, dataset_name=new_dataset_name)
+
+        # Get variables for old and new datasets.
+        old_variables = get_variables_in_dataset(db_conn=db_conn, dataset_id=old_dataset_id)
+        new_variables = get_variables_in_dataset(db_conn=db_conn, dataset_id=new_dataset_id)
+    
+    # Manually map old variable names to new variable names.
+    mapping = map_old_and_new_variables(
+        old_variables=old_variables, new_variables=new_variables, omit_identical=omit_identical,
+        matching_function=matching_function)
+
+    # Display summary.
+    display_summary(old_variables=old_variables, new_variables=new_variables, mapping=mapping)
+
+    # Save mapping to json file.
+    save_variable_replacements_file(mapping, output_file=output_file)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description="Map variable names from an old version of a dataset to variables of a new dataset."
+    )
+    parser.add_argument(
+        "-f",
+        "--output_file",
+        default=OUTPUT_FILE,
+        help=f"Path to output json file. Default: "
+        f"{OUTPUT_FILE}",
+    )
+    parser.add_argument(
+        "-old",
+        "--old_dataset_name",
+        help=f"Old dataset name (as defined in grapher).",
+    )
+    parser.add_argument(
+        "-new",
+        "--new_dataset_name",
+        help=f"New dataset name (as defined in grapher).",
+    )
+    parser.add_argument(
+        "-a",
+        "--add_identical_pairs",
+        default=False,
+        action="store_true",
+        help="If given, add variables with identical names in both datasets to the comparison. "
+             "If not given, omit such variables and assume they should be paired.",
+    )
+    args = parser.parse_args()
```

Comment:
Would be nice to have an `--interactive` option that is False by default. If False, then the fuzzy matching is executed and saved as a csv or json, with the expectation that the user will edit this file to confirm/reject each fuzzy match. If True, then the user is asked to confirm each match using the command line interface.

e.g. with `interactive=False`, the script could output a csv with columns `old_name,old_id,new_name0,new_id0,new_similarity0,...,new_name9,new_id9,new_similarity9` where each row is an old variable and its 10 best matches in the new dataset could be nice, as then it is easy to manually edit in vscode using multi-line editing, regex replace, etc as needed.

Main benefits of `interactive=False` would be that it makes it easier for the user to jump around (e.g. to change a fuzzy match that I skipped earlier) and to do the work in chunks (useful for large datasets like World Bank WDI with >1400 variables).

Reply:
Let's leave this for a future commit. For the moment, it's not a big deal to press enter for each variable. If there are thousands of variables, it's a bit annoying, but I suppose we can keep the enter key pressed for a few seconds :'-).